src/Microsoft.Health.Fhir.Shared.Api/Features/ActionResults/FhirResult.cs

```diff
@@ -46,6 +48,28 @@ public static FhirResult Create(ResourceElement resource, HttpStatusCode statusC
             };
         }
 
+        /// <summary>
+        /// Creates a FHIR result with the specified parameters
+        /// </summary>
+        /// <param name="bundleResponse">The resonse bundle.</param>
+        public static FhirResult Create(BundleResponse bundleResponse)
+        {
+            EnsureArg.IsNotNull(bundleResponse, nameof(bundleResponse));
+
+            if (bundleResponse.BundleProcessingStatus == BundleProcessingStatus.FAILED)
```

Comment:
Is this needed? If the operation failed, most likely you will be returning OperationOutcome instead of Bundle for transaction.

Reply:
This will not be needed. I have a updated PR that use OperationOutcome.